runtime/runtime/src/verifier.rs

```diff
@@ -545,6 +677,119 @@ mod tests {
         );
     }
 
+    mod zero_balance_account_tests {
+        use crate::near_primitives::account::id::AccountId;
+        use crate::near_primitives::account::{
+            AccessKeyPermission, Account, FunctionCallPermission,
+        };
+        use crate::verifier::is_zero_balance_account;
+        use crate::verifier::tests::{setup_accounts, TESTING_INIT_BALANCE};
+        use near_primitives::account::AccessKey;
+        use near_store::{get_account, TrieUpdate};
+        use testlib::runtime_utils::{alice_account, bob_account};
+
+        fn set_up_test_account(
+            account_id: &AccountId,
+            num_full_access_keys: u64,
+            num_function_call_access_keys: u64,
+            has_contract_code: bool,
+            has_contract_data: bool,
+        ) -> (Account, TrieUpdate) {
+            let mut access_keys = vec![];
+            for _ in 0..num_full_access_keys {
+                access_keys.push(AccessKey::full_access());
+            }
+            for _ in 0..num_function_call_access_keys {
+                let access_key = AccessKey {
+                    nonce: 0,
+                    permission: AccessKeyPermission::FunctionCall(FunctionCallPermission {
+                        allowance: Some(100),
+                        receiver_id: bob_account().into(),
+                        method_names: vec![],
+                    }),
+                };
+                access_keys.push(access_key);
+            }
+            let (_, state_update, _) = setup_accounts(vec![(
+                account_id.clone(),
+                TESTING_INIT_BALANCE,
+                0,
+                access_keys,
+                has_contract_code,
+                has_contract_data,
+            )]);
+            let account = get_account(&state_update, account_id).unwrap().unwrap();
+            (account, state_update)
+        }
+
+        #[test]
+        fn test_zero_balance_account_with_keys_and_contract() {
+            for num_full_access_key in 0..10 {
```

Comment:
add a comment mentioning that we're testing all the combinations of number of access keys